obspy/io/sac/sactrace.py

```diff
@@ -1092,7 +1092,7 @@ def read(cls, source, headonly=False, ascii=False, byteorder=None,
                 val = _ut._clean_str(val, strip_whitespace=False)
                 if val.startswith(native_str('-12345')):
                     val = HD.SNULL
-                hs[i] = val
+                hs[i] = val.encode('ASCII', 'replace')
```

Comment:
`_ut._clean_str` produces a `str`; `encode` goes from `str` to `bytes`; I think you might have something backwards. Are you testing on Python 3?